src/DIRAC/Resources/Computing/AREXComputingElement.py

```diff
@@ -39,335 +29,257 @@
-            print("Job submission status : ", r.json()['job']['status-code'], type(r.json()['job']['status-code']))
-            jobInfo = json.loads(r.text)['job']
-            if r.ok and jobInfo['status-code'] != '500':
-                # Job successfully submitted. Should I just test for == 201?
-                pilotJobReference = self._pilot_toAPI(jobInfo["id"])
-                batchIDList.append(pilotJobReference)
-                stampDict[pilotJobReference] = diracStamp
-                self.log.debug("Successfully submitted job", "%s to CE %s" % (pilotJobReference, self.ceHost))
-            else:
+            xrslString, diracStamp = self._writeXRSL(executableFile)
+            xrslString = xrslString + delegation
+            self.log.debug("XRSL string submitted", "is %s" % xrslString)
+            self.log.debug("DIRAC stamp for job", "is %s" % diracStamp)
+
+            # Submit the POST request
+            response = self.session.post(
+                query, data=xrslString, headers=self.headers, params=params, timeout=self.arcRESTTimeout
+            )
+            if not response.ok:
+                self.log.warn(
+                    "Failed to submit job",
+                    "to CE %s with error - %s - and messages : %s"
+                    % (self.ceHost, response.status_code, response.reason),
+                )
+                break
+
+            responseJob = response.json()["job"][0]
+            if responseJob["status-code"] > "400":
                 self.log.warn(
                     "Failed to submit job",
-                    "to CE %s with error - %s - and message : %s"
-                    % (self.ceHost, jobInfo['status-code'], jobInfo['reason']),
+                    "to CE %s with error - %s - and messages: %s"
+                    % (self.ceHost, responseJob["status-code"], responseJob["reason"]),
                 )
-                self.log.debug("DIRAC stamp and ARC job", "%s : %s" % (diracStamp, xrslString))
-                break  # Boo hoo *sniff*
+                break
+
+            jobID = responseJob["id"]
+            pilotJobReference = self._arcToDiracID(jobID)
+            batchIDList.append(pilotJobReference)
+            stampDict[pilotJobReference] = diracStamp
+            self.log.debug("Successfully submitted job", "%s to CE %s" % (pilotJobReference, self.ceHost))
+
+            # At this point, only the XRSL job has been submitted to AREX services
+            # Here we also upload the executable.
+            command = "jobs/" + jobID + "/session/" + os.path.basename(executableFile)
+            query = self.base_url + command
+
+            # Extract the content of the file
+            with open(executableFile, "r") as f:
+                content = f.read()
+
+            # Submit the PUT request
+            response = self.session.put(query, data=content, headers=self.headers, timeout=self.arcRESTTimeout)
```

Comment:
I am not sure what is the motivation of this. Could you explain please - Isn't it the job of DIRAC to actually download the executable or am I missing some point here?

Reply:
No - I have not observed this issue! In the test system the jobs seem to be running fine without it. I think this is not there in the ARCComputingElement either - even there, we just submit the XRSL string alone to the CE.

Reply:
> No - I have not observed this issue! In the test system the jobs seem to be running fine without it.

I don't understand how this could work. 
The request cannot find the executable and upload it automatically.
What kind of executable do you use in your XRSL file?

> I think this is not there in the ARCComputingElement - even there, we just submit the XRSL string alone to the CE.

Because in `ARCComputingElement` we are using an ARC client that takes care of this for us.
Here we have to do this manually.